### PR TITLE
feat: add support to specify tables in list of columns

### PR DESCRIPTION
**Pull request by imreACTmd**
Now you can replace columns in a specifiic table instead of all tables that have the same column name. You can also specify different replacements for the same column name in different table.

also fixed a bug where a static word replacement was overridden by faker.random.word.

Merry Christmas!

### CODE REVIEW

**Comment by imreACTmd** [comment]
Any comments @rap2hpoutre ?

File: README.md [hunk 1]
```diff
@@ -70,10 +76,6 @@ module.exports = {
 };
 ```
 
-### Memory limit
-
-Use `-m` to change `pg_dump` output memory limit (e.g: `512`)
-
```
Comment (rap2hpoutre) [inline README.md:76]: So, this is now obsolete, right?

File: src/index.ts [hunk 2]
```diff
@@ -127,9 +128,14 @@ class PgAnonymizer extends Command {
...
             if (indices.includes(k)) {
-              const replacement = list.find(
+              let replacement = list.find(
                 (l) => l.col === cols[k]
               )?.replacement;
+              if (!replacement) {
+                replacement = list.find(
+                  (l) => l.col === table + '/' + cols[k]
+                )?.replacement;
+              }
```
Comment (rap2hpoutre) [inline src/index.ts:138]: I can't understand my code anymore 😅 so I guess your implementation is OK then! I'm not sure I understand why there is a `if (!replacement)`, still it's the end of the year and I must be tired. Let's go!

File: src/index.ts [hunk 3]
```diff
@@ -148,18 +154,19 @@ class PgAnonymizer extends Command {
...
+                if (cols[k] === "name") return faker.name.findName();
+                if (cols[k] === "description") return faker.random.words(3);
+                if (cols[k] === "address") return faker.address.streetAddress();
+                if (cols[k] === "city") return faker.address.city();
+                if (cols[k] === "country") return faker.address.country();
+                if (cols[k] === "phone") return faker.phone.phoneNumber();
+                if (cols[k] === "comment") return faker.random.words(3);
+                if (cols[k] === "birthdate")
+                  return postgreSQLDate(faker.date.past());
+                return faker.random.word();
```
Comment (rap2hpoutre) [inline src/index.ts:168]: I originally put these lines out of else based on the "avoid else" statement I read here and there (it's about cyclomatic complexity). Not sure it's a piece of good advice though, so feel free to keep it like this!

**Review by rap2hpoutre — COMMENTED** [review]
Thank you (again) for your help and to keep this lib alive thanks to your work 🙏 

Just before I approve this PR, I want to be sure `/` is the "correct" character to specify tables names. In another (super similar) library I created for mongodb,  I choose to separate the collection from fields with dots: https://github.com/rap2hpoutre/mongodb-anonymizer#specify-list-of-fields-to-anonymize

```
users.email,products.price
```

So in our case it could be just:

```
public.user.email,public.product.description
```

And for a more complicated case using faker it could be:

```
public.user.firstName:faker.name.firstName
```

I would eventually prefer using this syntax because:
 - `/` could seems weird to users
 - `.` is more "classic" and feels like SQL
 - `.` is similar to what I implemented in this (copy/pasted) lib: https://github.com/rap2hpoutre/mongodb-anonymizer


What do you think?

**Comment by imreACTmd** [comment]
I thought that dots would be allowed in column names, but I reviewed the postgresql docs and they are not allowed, so I support switching from / to . and will do the change.

File: README.md (hunk 1 above)
Comment (imreACTmd) [reply README.md:76]: Correct. I'm debating leaving the option in so people's scripts don't break, what do you think? maybe a message saying this is obsolete?

**Comment by imreACTmd** [comment]
OK updated based on your comments.

**Review by rap2hpoutre — APPROVED** [review]
🎉  Thank you, you can merge it! I will release it on NPM right after (in the meanwhile, I guess I will add semantic release in order to automate the release process that should not rely only on my will).

_EDIT: I just added an issue here: https://github.com/rap2hpoutre/pg-anonymizer/issues/14_

**Comment by imreACTmd** [comment]
Let me look at issue #14 and add those. I have two more things I'd like to add - load the replacement list from a file, and send the output to stdout. Should I add that to this PR, or start a new one?

**Comment by rap2hpoutre** [comment]
I suggest we open one PR for each subject. I guess this PR is mergeable as-is then. Feel free to merge it now! (I can also merge it myself if you prefer!)

**Pull request merged**